### DISCUSSION TITLE
Keep getting error "Cannot set property '_x_dataStack' of null" when trying to use x-for

### DISCUSSION THREAD

**Discussion by mlab817** [alpinejs/alpine#1954] [1. Help]
So, I passed a Laravel variable into x-data as follows:

```
<div x-data="{ regionInfrastructures: [
            {"id":1, "region":"some text"}
      ] 
}">
      <template x-for="(region, index) in regionInfrastructures"></template>
</div>
```

Removed the rest of the code from above for brevity. 

But I keep getting the following error:

```
Alpine Expression Error: Cannot set property '_x_dataStack' of null

Expression: "regionInfrastructures"

 <template x-for=​"(region, index)​ in regionInfrastructures" :key=​"index">​…​</template>​
```

When I try to render the variable using x-text="JSON.stringify(regionInfrastructures)", it is returning the data. 

So what I am missing here?

**Comment by KevinBatdorf** [comment]
Can you provide a real example on CodePen or something similar? That error will show with the above code because there's nothing inside `<template></template>`

**Reply by mlab817** [reply]
here: https://codepen.io/mlab817/pen/xxdevqd

Now that I see it, it seems that there's something wrong with the data being passed since it's json data. by the way, the data is passed using laravel.

**Reply by KevinBatdorf** [reply]
That example also has nothing between the `<template></template>` tags. 

>  it seems that there's something wrong with the data being passed

You have unescaped strings is all.

**Reply by mlab817** [reply]
Ok. thanks. I missed that.

**Reply by devmount** [reply]
Just for reference: I had the same error when putting more than one elements into the `<template>` (which is not allowed but I wasn't aware 😇 )

**Reply by ekwoka** [reply]
Yes, templates need to have a single node.

**Reply by devmount** [reply]
I know that now, but it isn't obvious given the error message above.

**Comment by mlab817** [comment]
Seems that my problem was totally unrelated to the data. I made a mistake in binding the name attribute. So i changed it from:

```
x-bind:name="region_infrastructures[${index}][y2021]"
```

to:

```
x-bind:name="`region_infrastructures[${index}][y2021]`"
```

I missed the backticks
